audio.asm

```diff
@@ -352,248 +352,27 @@ INCLUDE "audio/sfx/cry21_3.asm"
-	ld de, Music_MeetRival_branch_b21d
-	call Audio1_OverwriteChannelPointer
-	ld de, Music_MeetRival_branch_b2b5
-
-Audio1_OverwriteChannelPointer:
-	ld a, e
-	ld [hli], a
-	ld a, d
-	ld [hli], a
-	ret
-
-; an alternate tempo for MeetRival which is slightly slower
-Music_RivalAlternateTempo::
-	ld c, BANK(Music_MeetRival)
-	ld a, MUSIC_MEET_RIVAL
-	call PlayMusic
-	ld hl, wChannelCommandPointers
-	ld de, Music_MeetRival_branch_b119
-	jp Audio1_OverwriteChannelPointer
-
-; applies both the alternate start and alternate tempo
-Music_RivalAlternateStartAndTempo::
-	call Music_RivalAlternateStart
-	ld hl, wChannelCommandPointers
-	ld de, Music_MeetRival_branch_b19b
-	jp Audio1_OverwriteChannelPointer
-
-; an alternate tempo for Cities1 which is used for the Hall of Fame room
-Music_Cities1AlternateTempo::
-	ld a, 10
-	ld [wAudioFadeOutCounterReloadValue], a
-	ld [wAudioFadeOutCounter], a
-	ld a, $ff ; stop playing music after the fade-out is finished
-	ld [wAudioFadeOutControl], a
-	ld c, 100
-	call DelayFrames ; wait for the fade-out to finish
-	ld c, BANK(Music_Cities1)
-	ld a, MUSIC_CITIES1
-	call PlayMusic
-	ld hl, wChannelCommandPointers
-	ld de, Music_Cities1_branch_aa6f
-	jp Audio1_OverwriteChannelPointer
+INCLUDE "audio/alternate_tempo.asm"
 
 
 SECTION "Audio Engine 2", ROMX
 
-Music_DoLowHealthAlarm::
-	ld a, [wLowHealthAlarm]
-	cp $ff
-	jr z, .disableAlarm
-
-	bit 7, a  ;alarm enabled?
-	ret z     ;nope
-
-	and $7f   ;low 7 bits are the timer.
-	jr nz, .asm_21383 ;if timer > 0, play low tone.
-
-	call .playToneHi
-	ld a, 30 ;keep this tone for 30 frames.
-	jr .asm_21395 ;reset the timer.
-
-.asm_21383
-	cp 20
-	jr nz, .asm_2138a ;if timer == 20,
-	call .playToneLo  ;actually set the sound registers.
-
-.asm_2138a
-	ld a, $86
-	ld [wChannelSoundIDs + Ch5], a ;disable sound channel?
-	ld a, [wLowHealthAlarm]
-	and $7f ;decrement alarm timer.
-	dec a
-
-.asm_21395
-	; reset the timer and enable flag.
-	set 7, a
-	ld [wLowHealthAlarm], a
-	ret
-
-.disableAlarm
-	xor a
-	ld [wLowHealthAlarm], a  ;disable alarm
-	ld [wChannelSoundIDs + Ch5], a  ;re-enable sound channel?
-	ld de, .toneDataSilence
-	jr .playTone
-
-;update the sound registers to change the frequency.
-;the tone set here stays until we change it.
-.playToneHi
-	ld de, .toneDataHi
-	jr .playTone
-
-.playToneLo
-	ld de, .toneDataLo
-
-;update sound channel 1 to play the alarm, overriding all other sounds.
-.playTone
-	ld hl, rNR10 ;channel 1 sound register
-	ld c, $5
-	xor a
-
-.copyLoop
-	ld [hli], a
-	ld a, [de]
-	inc de
-	dec c
-	jr nz, .copyLoop
-	ret
-
-;bytes to write to sound channel 1 registers for health alarm.
-;starting at FF11 (FF10 is always zeroed), so these bytes are:
-;length, envelope, freq lo, freq hi
-.toneDataHi
-	db $A0,$E2,$50,$87
-
-.toneDataLo
-	db $B0,$E2,$EE,$86
-
-;written to stop the alarm
-.toneDataSilence
-	db $00,$00,$00,$80
-
-
-INCLUDE "engine/menu/bills_pc.asm"
-
+INCLUDE "audio/low_health_alarm.asm"
+INCLUDE "engine/pokemon/bills_pc.asm"
```

Comment:
Now that linkerscripts are a thing, can this include be moved from audio.o to main.o and still link properly?

Reply:
Yes it can.